### ISSUE TITLE
Cache management breaks shared installs of JBang in Linux

### ISSUE THREAD

**Issue by hguerrero** [jbangdev/jbang#1638]
**Describe the bug**

In v0.103 a new cache management feature was included to validate the freshness of files downloaded by JBang. To check the latest download, it implemented `Files.setLastModifiedTime` among other validations. 

However, this implementation breaks shared installations of JBang as the Java method requires that the owner of the file is the only one allowed to change the file. Hence getting an `Operation Not Permitted` error.

**To Reproduce**
1. Create two users (user1,user2)
2. Assign the same $HOME to both users say `/home/user`
3. Install and setup the app (camel@apache/camel) using user1
4. Log in as user2 and try to run `camel init hello.js`

**Expected behavior**
As it worked in 0.102 the application succeeds in creating the file. However you will get the following error:

```
[jbang] [ERROR] Could not download https://github.com/apache/camel/blob/HEAD/jbang-catalog.json
[jbang] Run with --verbose for more details
```

**JBang version**
Cache: /home/user/.jbang/cache
Config: /home/user/.jbang
Repository:/home/user/.m2/repository
0.108.0

**Additional context**
References:
https://stackoverflow.com/questions/61011124/java-file-setlastmodified-fails-on-linux-when-you-are-not-the-file-owner

**Comment by maxandersen** [comment]
Not saying we can't find ways to gracefully handle this but jbangs cache is supposed to be per user. 

Struggling to see how this is supposed to work if there are concurrent runs from different users at the same time ?

**Comment by quintesse** [comment]
Yeah, this might have worked for your particular use-case, but I see a whole bunch of possible situations where things could go horribly wrong. Having two users share a single home directory seems a recipe for disaster TBH.

**Comment by hguerrero** [comment]
A particular use case is related to the use of JBang with containers. When installing an app during the container image build, we can have a specific user provide the current home directory for the installation. 

However, if that container runs on particular platforms, we can't guarantee that the runAsUser is the same one we used for the image build. Adding them to the same group, like `0`, allows us to share the directories and files installed, but with v0.103+, we face the issue with the cache and the owner restrictions. 

Technically you end with 2 users sharing the same filesystem where both of them have group permissions, but only one is the owner. You can't do any further modifications as the running container doesn't support `su` or `sudo`.

**Comment by quintesse** [comment]
Btw @hguerrero , would you be so kind as to copy&paste the actual error? (by running with `--verbose`). Thanks.

I think we should at least capture the exception and simple ignore it (write some debug message). That way things will at least work for you, although it will mean files will get re-downloaded each time.

In the meantime try setting `cache-evict` to `never` by running `jbang config set cache-evict never`, although this will prevent downloads from getting updated it should at least prevent the error from occurring.

**Comment by hguerrero** [comment]
Workarounds we have found are:

- `jbang config set cache-evict never`
- `jbang cache clear`

Both of them skip the issue.

**Comment by quintesse** [comment]
Ok, created a fix #1639 which should appear in the next release
